meteor/client/ui/App.tsx

```diff
@@ -134,23 +176,38 @@ class App extends React.Component<InjectedI18nProps, IAppState> {
 							<Route path='/countdowns/presenter' component={NullComponent} />
 							<Route path='/activeRundown' component={NullComponent} />
 							<Route path='/prompter/:studioId' component={NullComponent} />
-							<Route path='/' render={(props) => <Header {...props} allowConfigure={this.state.allowConfigure} allowTesting={this.state.allowTesting} allowDeveloper={this.state.allowDeveloper} />} />
+							<Route path='/' render={(props) => <Header 
+								{...props}
+								userId={this.props.userId}
+								allowConfigure={this.state.allowConfigure} 
+								allowTesting={this.state.allowTesting} 
+								allowDeveloper={this.state.allowDeveloper} 
+							/>} />
 						</Switch>
 					</ErrorBoundary>
 					{/* Main app switch */}
 					<ErrorBoundary>
 						<Switch>
-							{/* <Route exact path='/' component={Dashboard} /> */}
-							<Route exact path='/' component={RundownList} />
-							<Route path='/rundowns' component={RundownList} />
-							<Route path='/rundown/:playlistId' component={RundownView} />
-							<Route path='/activeRundown/:studioId' component={ActiveRundownView} />
-							<Route path='/prompter/:studioId' component={PrompterView} />
-							<Route path='/countdowns/:studioId/presenter' component={ClockView} />
-							<Route path='/status' component={Status} />
-							<Route path='/settings' component={Settings} />
+							{Settings.enableUserAccounts ?
+								[
+									<Route key='0' exact path='/' component={(props) => <LoginPage {...props}/>} />,
```

Comment:
I would suggest not using arrays, if you are not actually providing an array of items. Use a <React.Fragment> or the shorthand `<></>` as that will perform way better and you don't have to set any dummy `key` attributes.

Reply:
I originally had a fragment but when the `Settings.enableUserAccounts` returns it, the Switch thinks it is the component to render instead of checking the sub-components.